etcdserver/api/membership/cluster.go

```diff
@@ -765,10 +765,17 @@ func ValidateClusterAndAssignIDs(lg *zap.Logger, local *RaftCluster, existing *R
 	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
 	defer cancel()
 	for i := range ems {
-		if ok, err := netutil.URLStringsEqual(ctx, lg, ems[i].PeerURLs, lms[i].PeerURLs); !ok {
+		var err error
```

Comment:
@dbavatar I don't have a way to reproduce the error as you see with your config. However, I tested your fix. We probably don't need to sort anymore with your proposed fix. I am not comfortable on suggesting to change the current pattern but let's wait on @hexfusion for this thoughts. About test case, adding a new case in netutil_test for urlsEqual similar to what's there covering your use case should a good add. Thanks!